packages/x-data-grid/src/components/cell/GridCell.tsx

```diff
@@ -350,17 +332,12 @@ const GridCell = React.forwardRef<HTMLDivElement, GridCellProps>(function GridCe
       ...styleProp,
     } as React.CSSProperties;
 
-    const isLeftPinned = pinnedPosition === PinnedPosition.LEFT;
-    const isRightPinned = pinnedPosition === PinnedPosition.RIGHT;
-
-    if (isLeftPinned || isRightPinned) {
-      let side: 'left' | 'right' = isLeftPinned ? 'left' : 'right';
-
-      if (isRtl) {
-        side = isLeftPinned ? 'right' : 'left';
-      }
+    const isLeftPinned = pinnedPosition === PinnedColumnPosition.LEFT;
+    const isRightPinned = pinnedPosition === PinnedColumnPosition.RIGHT;
 
-      cellStyle[side] = pinnedOffset;
+    const pinnedSide = rtlFlipSide(pinnedPosition, isRtl);
+    if (pinnedSide && pinnedOffset !== undefined) {
```

Comment:
Could that condition be expressed in terms of `pinnedPosition` instead? And would that allow us to keep the type of `pinnedOffset: number` instead of `?: number`? And could we move the flip logic inside the if check?

Reply:
I actually did it like that initially, but it felt wrong to me. Why create a value that shouldn't exist – could end up in adding accidental inline styles for non-pinned columns that should never exist. The type of `pinnedOffset` could be constrained to only exist if `pinnedPoisition` is left/right though.

Moving the rtlFlipSide could make sense, albeit it currently also narrows the type of pinnedPosition – if pinnedSide is defined, pinnedPosition can only be left or right, so maybe not much of a change:
<img width="740" alt="Screenshot 2024-12-03 at 00 26 17" src="https://github.com/user-attachments/assets/90b93d00-95f1-474e-ba81-f30ee2f5956e">